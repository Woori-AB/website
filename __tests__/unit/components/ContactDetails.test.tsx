import { cleanup, render } from "@testing-library/react";
import { describe, it, vi, expect, afterEach } from "vitest";

describe("ContactDetails", () => {
  afterEach(() => {
    cleanup();
    vi.clearAllMocks();
  });

  it("matches snapshot(s)", async () => {
    // Arrange
    const { default: ContactDetails } = await vi.importActual<
      // eslint-disable-next-line @typescript-eslint/consistent-type-imports
      typeof import("@/components/ContactDetails")
    >("@/components/ContactDetails");

    // Act
    const { container } = render(<ContactDetails />);

    // Assert
    expect(container).toMatchSnapshot();
  });
});
