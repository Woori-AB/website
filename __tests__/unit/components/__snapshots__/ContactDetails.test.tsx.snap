// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ContactDetails > matches snapshot(s) 1`] = `
<div>
  <div
    class="flex flex-col items-center text-center gap-4"
    id="contact-details"
  >
    <div
      class="flex flex-col items-center text-center"
    >
      <h1>
        mocked-i18n-'visit-us'
      </h1>
      <span>
        <a
          class="flex gap-1 hover:underline items-start text-left"
          href="https://maps.app.goo.gl/9Ne5EDEXKdsdvig7A"
        >
          <mocked-lucide-react-mappin
            size="1.2rem"
          />
          Stagneliusv√§gen 47, 112 57, Stockholm
        </a>
      </span>
    </div>
    <div
      class="flex flex-col items-center text-center"
    >
      <h1>
        mocked-i18n-'contact'
      </h1>
      <span>
        <a
          class="flex gap-1 hover:underline items-center"
          href="tel:+4686566675"
        >
          <mocked-lucide-react-phone
            size="1.2rem"
          />
          08-656 66 75
        </a>
      </span>
      <span>
        <a
          class="flex gap-1 hover:underline items-center"
          href="mailto:kontakt@wooristockholm.se"
        >
          <mocked-lucide-react-mail
            size="1.2rem"
          />
          kontakt@wooristockholm.se
        </a>
      </span>
    </div>
  </div>
</div>
`;
